src/components/Header.jsx

```diff
@@ -1,29 +1,46 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Link from 'next/link';
-import PropTypes from 'prop-types';
 import styled from '@emotion/styled';
 
-export default function Header({ openModal, loginStatus }) {
+import CreateChannel from './Main/CreateChannel';
+import Modal from './Modal';
+
+export default function Header() {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isLogin, setIsLogin] = useState(false);
+
+  useEffect(() => {
+    setIsLogin(!!sessionStorage.getItem('user'));
+  });
+
+  function openModal() {
+    setIsModalOpen(true);
+  }
+
+  function closeModal() {
+    setIsModalOpen(false);
+  }
+
   return (
-    <HeaderWrapper loginStatus={loginStatus}>
+    <HeaderWrapper loginStatus={isLogin}>
       <Link href="/main" passHref>
         <span className="nav">CHANNEL</span>
       </Link>
       <Link href="/" passHref>
         <span className="nav logo">VLIVE</span>
       </Link>
-      <span href="/channel" className="nav create-channel" onClick={openModal}>
+      <span className="nav create-channel" onClick={openModal}>
         CREATE CHANNEL
       </span>
+      {isModalOpen && (
+        <Modal closeModal={closeModal}>
+          <CreateChannel closeModal={closeModal} isModalOpen={isModalOpen} />
```

Comment:
로직이 간단하니 함수를 인라인으로 작성해서 내려줘도 될 것 같고, prop 네이밍을 수정하면 좋을 것 같아요!
isModalOpen은 CreateChannel컴포넌트가 보여지냐 안보여지냐를 결정하는 boolean값이니까 show나 open도 괜찮을 것 같습니다.

```suggestion
          <CreateChannel onClose={closeModal} open={isModalOpen} />
```